juriscraper/pacer/appellate_attachment_page.py

```diff
@@ -0,0 +1,217 @@
+        :return: main_document_id
+        """
+        try:
+            rows = self.tree.xpath("//tbody/tr/td/a/@href")
+            pre_doc = sorted(row.split("/")[-1] for row in rows)[0]
+            return f"{pre_doc[:3]}0{pre_doc[4:]}"
+        except IndexError:
+            return None
+
+    def _get_attachment_number(self, row) -> int:
+        """ "Return the attachment number for an item.
+
+        :param row: Table row as an lxml element
+        :type: lxml HTML element
+        :return: Attachment number for row
+        :type: int
+        """
+        number = int(row.xpath(".//td/text()")[0].strip())
+        return number
+
+    def _get_description_from_tr(self, row):
+        """Get the description from the row
+
+        :param row: Table row
+        :type: lxml HTML element
+        :return: Row description
+        :type: unicode
+        """
+        description_text_nodes = row.xpath(".//td")
+        if not description_text_nodes:
+            return ""
+        return force_unicode(description_text_nodes[2].text_content().strip())
+
+    @staticmethod
+    def _get_page_count_from_tr(row):
+        """Take a row from the attachment table and return the page count as an
+        int extracted from the cell specified by index.
+        """
+        description_text_nodes = row.xpath(".//td")
+        if not description_text_nodes:
+            return None
+        return int(description_text_nodes[3].text_content().strip())
+
+    @staticmethod
+    def _get_pacer_doc_id(row):
+        """Take in a row from the attachment table and return the pacer_doc_id
+        for the item in that row. Return None if the ID cannot be found.
+        """
+        try:
+            url = row.xpath(".//a")[0]
+        except IndexError:
+            # Item exists, but cannot download document. Perhaps it's sealed
+            # or otherwise unavailable in PACER. This is carried over from the
+            # docket report and may not be needed here, but it's a good
+            # precaution.
+            return None
+        else:
+            doc1_url = url.xpath("./@href")[0]
+            return get_pacer_doc_id_from_doc1_url(doc1_url)
+
+    def _get_pacer_case_id(self):
+        """Get the pacer_case_id value by inspecting the function scripts
+
+        :returns str: The pacer_case_id value
+        """
+        script_html = html.tostring(self.tree.xpath(".//script")[0]).decode()
+        m = re.search(r"[?&]caseid=(\d+)", script_html, flags=re.I)
+        if m:
+            return m.group(1)
+
+    def _get_pacer_seq_no(self):
+        """Get pacer sequence number.
+
+        This value corresponds to the value in the docket TR
+        """
+        script_html = html.tostring(self.tree.xpath(".//script")[0]).decode()
+        m = re.search(r"[?&]d=(\d+)&outputForm", script_html, flags=re.I)
+        if m:
+            return m.group(1)
+
+
+def _main():
```

Comment:
Why have a `_main()` function when you have a main() function? Seems weird and unnecessary?

Reply:
you know - maybe I should've just rewritten this.

Reply:
Changed to `main()`